src/BlobField/blobs.js

```diff
@@ -0,0 +1,210 @@
+		let easeFactor = this.defaultEasing;
+		if (this.isSlowSim) {
+			easeFactor = this.slowSimEasing;
+		}
+		const dx = this.mouseTargetX - this.mouseCurrX;
+		this.mouseCurrX += dx * easeFactor;
+		const dy = this.mouseTargetY - this.mouseCurrY;
+		this.mouseCurrY += dy * easeFactor;
+
+		// Move the hidden/mouse blob
+		if (this.collapsed) {
+			this.balls[0].point = new paper.Point(
+				-this.balls[0].radius,
+				-this.balls[0].radius
+			);
+		} else {
+			this.balls[0].point = new paper.Point(this.mouseCurrX, this.mouseCurrY);
+		}
+	}
+
+	onResize() {
+		for (let i = 0; i < this.balls.length; i++) {
+			this.balls[i].radius = this.calcRadius(i);
+			let tempIsVert = paper.view.size.width > paper.view.size.height;
+			this.balls[i].isVertical = tempIsVert;
+		}
+		this.balls[0].radius = this.calcRadius(0) * this.mouseRadiusMultiplier;
+	}
+
+	onMouseMove(x,y) {
+		this.mouseTargetX = x;
+		this.mouseTargetY = y;
+	}
+
+	pathOnMouseEnter(event) {
+		this.isSlowSim = true;
+
+		const idx = event.target.idx;
+		this.balls[idx].mouseEnterPt = event.point;
+
+		for (let i = 1; i < this.balls.length; i++) {
+			if (this.balls[i].path !== event.target) {
+				this.balls[i].path.tween({
+						opacity: this.balls[i].path.opacity,
+						shadowColor: this.balls[i].path.shadowColor,
+					}, {
+						opacity: this.hoverFadedOpacity,
+						shadowColor: this.balls[i].shadowInactiveColor,
+					},
+					this.animationDuration
+				);
+			} else {
+				this.balls[i].path.tween({
+						opacity: this.balls[i].path.opacity,
+						// shadowColor: this.balls[i].path.shadowColor,
+					}, {
+						opacity: this.hoverActiveOpacity,
+						// shadowColor: this.balls[i].shadowColor,
+					},
+					this.animationDuration
+				);
+			}
+    }
+    this.onArtistHovered && this.onArtistHovered(event.target.artist);
+	}
+
+	pathOnMouseLeave(event) {
+		this.isSlowSim = false;
+
+		const idx = event.target.idx;
+		this.balls[idx].mouseLeavePt = event.point;
+
+		// Add force on mouse leave
+		// this.repulseBall(idx);
+
+		for (let i = 1; i < this.balls.length; i++) {
+			this.balls[i].path.tween({
+					opacity: this.balls[i].path.opacity,
+					shadowColor: this.balls[i].path.shadowColor,
+				}, {
+					opacity: this.opacity,
+					shadowColor: this.balls[i].shadowInactiveColor
+				},
+				this.animationDuration
+			);
+		}
+
+    this.onArtistHovered && this.onArtistHovered();
+	}
+
+	repulseBall(idx) {
+		if (!this.collapsed) {
+			const repulsionV = this.balls[idx].mouseEnterPt.subtract(
+				this.balls[idx].mouseLeavePt
+			);
+			this.balls[idx].vector = this.balls[idx].vector.add(repulsionV.normalize());
+			this.balls[idx].vector = this.balls[idx].vector.subtract(this.balls[idx].radius);
+		}
+	}
+
+	pathOnClick(event) {
+		this.collapsed = !this.collapsed;
+		for (let i = 1; i < this.balls.length; i++) {
+			this.balls[i].radius = this.calcRadius(i);
+    }
+    // TODO make this callback work
+    //this.onArtistClicked && this.onArtistClicked(artist)
```

Comment:
how can we get a handle to current artist here?

Reply:
event.target.artist

Reply:
👍